### DISCUSSION TITLE
Skill tree REVAMP 2.0

### DISCUSSION THREAD

**Discussion by TheDcol** [sunflower-land/sunflower-land#3202] [Ideas]
The following is a continuation and summary of feedback received from: https://github.com/sunflower-land/sunflower-land/discussions/2180

Adam's feedback: https://github.com/sunflower-land/sunflower-land/discussions/2180#discussioncomment-5084747

The premise remains the same. Players will have their Bumpkin skills completely reset and depending on their level, players will be given a new amount of points to re-level the bumpkin. 1 level = 1 point. A lvl 55 bumpkin will be given 55 points. Everytime a Bumpkin levels up, they get 1 Skill Point.

The skill tree has more variety, options and specialisation opportunities. It is built as such that no player can (for now and for a while) have every skill unlocked at once. Every high level player has the same skills and specialisation only differs by the types of items they have. Bumpkin lock mechanisms that are currently present would function similarly to prevent abuse.

Below is an UNOFFICIAL skill tree proposition:

![image](https://github.com/sunflower-land/sunflower-land/assets/108110065/dfe864d7-cfb8-4aaf-956f-ff13cf537490)

Old Minerals Skill Tree layout
![image](https://github.com/sunflower-land/sunflower-land/assets/108110065/f084e3a2-4d6c-47a0-830b-aa064c965608)

New Minerals SKill Tree layout
![image](https://github.com/sunflower-land/sunflower-land/assets/108110065/0d11396e-af58-4466-8cb2-cb6647300f13)

As new features/mechanics come into place, we could easily add skills in the appropriate tree and row. I think we should make some sort of requirements to unlock skills in the Tier 3 rows (Maybe as such that a bumpkin needs at least 2 Tier 2 skills in the Wood tree to unlock the 3rd tier...)

If you have any thoughts, opinions or suggestions for this please post below (I saw mention of a General/QOL skill tree in the past) :)
Would also appreciate any new ideas for skills that you guys might have :D

Note: Red text would require some work beforehand, meaning the item/functionality is not yet ready

This is NOT final NOR would I say it is currently balanced, but I want to get the ball rolling again and get this out sooner than later if the community is for it.

Cheers!

**Comment by vitt0** [comment]
I like most of the skill and tree path, some small feedback for some that call my attention that I would change would be:

Fruit tree/bushes drop +0.2 wood
For a "tier 2 skill" this sounds pretty week Imo, If I have to count the fruit trees that I cut daily it would probably only be 1-3 Ig and I have 11 fruit patches so is like +0.2-0.6 wood really small buff, I think I would change it to maybe be "Fruit tree/bushes are considered as normal tree" so whatever other buff I have it would applied to them or something like that.

But tbh even that sounds kinda like a weak buff for a tier 2 skill, I think i would scrapt that idea and use something else, a cool idea could be a "% chance of that you cut the tree without using a axe" (downside it would be useless for Foreman owners) no idea about what % would be balance but sounds like a more interesting skill Imo

And the other suggestion would be to change the wording for Corale/Tango to just be in general like "50% profits on Fruit deliveries" Imo dosn't sound good to limit the skill to just X NPC, should be something that works for everything, even if in practice Corale is the only NPC that ask for fish deliveries I still think it looks better to describe it as a "general buff"

**Comment by brian-man** [comment]
A reminder to consider trade-off skills:

> Fertiliser Specialist
> Boost: +1 compost yield
> Penalty: -50% compost effect

If you use your compost, that hurts.  If you sell it, it's awesome maybe.

> Skill: Iron Idolatry
> Boost: +1 Iron yield
> Penalty: -0.5 Stone yield

> Skill: King Kale
> Boost: +0.2 Kale
> Penalty: -0.1 all other crops

**Comment by brian-man** [comment]
Thoughts on the sample image:
- Some details on the tiers would be helpful for modeling/evaluating.  Can I get all 3 tier 1 crops? How many of which tiers do I need to unlock one or more tier 4 of crops?
- "X% faster crop growth" / "grow X% faster" => "reduced crop growth duration"
- "AEO" => "AOE"
- Nit: For resource references, be consistent on singular or plural.
- "Profit + 20%" doesn't really work (as an equation) with the cost and yield varying across players.
- Beg: Include the numeric values in all "chance" descriptions.
- For the scarecrows, consider (inexpensive?) options to change the AOE position relative to the 'crow.
- Beg: "Unlock special crops" should have a formal statement from the dev team on whether these crops are part of seasonal chores/deliveries.
- Clarification: For "seed bonus from chest" are you referring to the RNG "crop rewards"?
- Concern: "Trees grow 10% quicker" - The existing tree duration boosts are getting close to flipping some users to negative times.
- Recommend: Fruit bushes - option to "discard" bushes without collecting the wood but not needing Axe.
- Clarification: Be clear whether it's 3x or +2.
- Recommend: "Lumber Jack building" - Use a different name to avoid confusion with legacy farm skill and current bumpkin skill.
- Beg: "Lumber Jack" building - Tools for Stone, Iron, and Gold need love too.
- Recommend: "Once every 10 hours" is 2x a day and also a weird clock for people to manage.  Recommend once (twice?) per daily reset.  Simpler to implement; model on 30 fishing casts.
- Clarification: Can all 4 bait fishing skills be claimed? Does that net +2 to all worm yield?
- :heart: "No treasures, only fish"
- Consider: "+.1 Egg Yield" - Change to +2 Egg Yield to first Chicken fed "today" (UTC).  For low and middle farms, this would be huge as it helps deal with low wheat supply.  For top farms no difference from what you have.
- Consider: ^ similar but for Stone, Iron, Gold yield.
- Consider: Queue multiple items to cook per building.  Same cooking rate, but like 10 Mashed Potato for 130 seconds is huge QoL update vs. 3 clicks every 13 seconds.
  - If that's "hard" to implement, then offer a "cook 10" button similar to eat 10.  Use 10 ingredients and duration 10x as long.

**Reply by TheDcol** [reply]
Thanks, I will take some of this feedback along with others and make some changes over the weekend <3

**Reply by TheDcol** [reply]
Regarding cooking multiple foods at once, I think that would best be done from a building upgrade option, what do you think? 

Lumberjack gives extra axes only, the Goblin Toolshed gives extra pickaxes (all types, including stone and iron pickaxes)

Regarding concern around 'trees grow 10% quicker' that is the current skill in the game, not a skill on top of. Aware that tree time is getting close though.. no more should be added)

Changed profit to revenue, fixed up some spelling and clarified some of the chances. Also added trade-off skills (blue text).

Tiers are currently 'open'. A player would only need to select one skill in tier 1 to unlock all in tier 2, and one skill in tier 2 to unlock all tier 3. Open to making it more complex/restrictive if the UI won't be messy.

All skills can stack IMO. 

![image](https://github.com/sunflower-land/sunflower-land/assets/108110065/3600d4d6-3572-4b9d-814c-66671130efc2)

Just throwing ideas on the board, want to keep refining it every so often

**Reply by TheDcol** [reply]
Drafted this for now
![image](https://github.com/sunflower-land/sunflower-land/assets/108110065/9f5ce435-95c6-4fb0-acd0-2dbc33d0a33a)

It goes to show how 'easy' it would be to add new skill trees. If someone runs out of skill points and a new tree is released that they desperately want, they will either need to reset their points (SFL fee? with a reset max every 3-6 months?) or they will need to eat more food/upgrade to unlock more points

**Reply by Bacardy2069** [reply]
I'd like to add my thoughts


I would like this to look more interesting than boring pumping


For example
20lvl 1-tier
40lvl 2-tier
60lvl 3-tier
 open access to 3 levels of professions and you need to do 3 quests for each profession

for a person to concentrate
on a specific
if he is a fisherman in life, he only wanted to fish for fish or a cook and will become a cook

so that there are no jumps from skill branches

make it unique and interconnected for everyone, so that there are active and passive skills for certain professions and recipes for everyone too, so that these items are required for other players, it is imperative that there is balance in the market

I have a lot of thoughts that I can’t put into a couple of sentences, I tried to briefly describe them.

**Reply by RicardoKeita** [reply]
> Drafted this for now ![image](https://private-user-images.githubusercontent.com/108110065/303950624-9f5ce435-95c6-4fb0-acd0-2dbc33d0a33a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA0NDA1NTgsIm5iZiI6MTcyMDQ0MDI1OCwicGF0aCI6Ii8xMDgxMTAwNjUvMzAzOTUwNjI0LTlmNWNlNDM1LTk1YzYtNGZiMC1hY2QwLTJkYmMzM2QwYTMzYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwOFQxMjA0MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYTY2ZGMyMDE2OWI3NjkwYTI5NzhjNjdjMzMyMDcyZmUzMzJmMGVmMDBhMzg3ODg1OGU2MzAxM2FiODkzZmM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Erk88VVq2qEIXgMHWLCalhvTI4uSAd2VDD_-yHMUgPk)
> 
> It goes to show how 'easy' it would be to add new skill trees. If someone runs out of skill points and a new tree is released that they desperately want, they will either need to reset their points (SFL fee? with a reset max every 3-6 months?) or they will need to eat more food/upgrade to unlock more points

Garanteed bee swams will invalidate the currently Bee Mask NFT, i think doing some ex.: +0.2 in Bee Swam Buff would be a better approach, valuing the NFT instead of invalidating it.

**Comment by Ivan-Ricardo-Lopes** [comment]
Instead of 1 skill per level, 1 skill point every two levels would enhance specialization. With this, a level 60 character would have a 100% proficiency in one skill tree, allocating only a few points to another.

**Reply by brian-man** [reply]
> Instead of 1 skill per level, 1 skill point every two levels would enhance specialization.

I prefer the 1 per level change as proposed.  Never answering "I levelled how come no skill point" in Discord ever again would be lovely.

**Reply by InuBakaBo** [reply]
Considering the numbers of different trees, I believe it would be better if people can spread their eggs into multiple baskets.

**Comment by InuBakaBo** [comment]
Can we also have reset feature which would cost SFL/resources? This would also allow some burning when people try to change their strategy with each season/event.

**Reply by TheDcol** [reply]
Yup forgot to include this in the post but it would certainly be good to have

**Comment by Thebyber** [comment]
Mushrooms are the forgotten ones in the game, they have very little drop and every 16 hours... they have no attractive utility. Why not implement a skill tree for this resource to give it a more attractive use? Here's my suggestion!

![image](https://github.com/sunflower-land/sunflower-land/assets/124185010/ffbe9b58-2e89-4719-86ff-0af14c7bf2af)

The idea is to give value to the mushroom by giving more drop and giving more experience to its recipes that have become obsolete. I would implement the mushroom trade in the bumpkin plaza.

With this new update on the prestige island there will be new resources, I came up with the idea of implementing the black truffle, which is a type of mushroom that is highly valued for cooking and very expensive.

The recipes that come to mind are (I'm not good at making recipes):
-Blue marvin truffled with vegetables: x2 Blue marvin + black truffle + x50 cabbage + x50 carrots (Kitchen) (+3000 exp)
-Truffle scrambled eggs: x20 Eggs + x20 potatoes + black truffle (Fire pit) (+2000 exp)

The experience of these foods can be changed to give more or less experience.

And another suggestion for the fishing branch would be to unlock a building to make sushi, you could make various dishes and you could also put crabs, seaweed, cucumber, pipi... the ideal would be to implement rice to make this building and have it have more sense.

I hope you like it!

**Reply by TheDcol** [reply]
#Justice4Mushrooms

Good post and good point!

Not sure what the team is thinking in relation to magic mushrooms yet.. 

Rice as a crop would be interesting..

**Comment by NotKeen** [comment]
Kind of a crazy idea but: Maybe we should just get rid of the concept of skills and do everything via NFTs / SFTs combined with Bumpkin level requirements for using them. My thinking is that it might be easier to design it that way, since skill trees tend to become outdated / irrelevant with changing metas, so doing it all via SFTs could give more flexibility / agility. @TheDcol 

As a transition everyone who has unlocked skills right now could get airdropped a set of SFTs that gives them the same boost (since it's often bad to take something away from people).

Edit: While tokenizing skills we could also do the same with the legacy skills (maybe with the exception of the Liquidity Provider badge) to make them tradable. I think it would be good to allow new people to buy them if they really want to. And also it would take demand away from the trading of whole farms, since currently the only way to attain the OG boosts is buying old farms.

**Reply by TheDcol** [reply]
This is an interesting approach, I wonder what @adamhannigan thinks

**Reply by rodolfoddias** [reply]
I have wrote a comment supporting this idea but after thinking out of it for a while, I believe it would turn things too complex. I think it's easier to just keep skills in the "web2" part of the game. People would probably try to exploit it with multi-accs to get different skills using less exp and it would force the team to build a mechanism just to prevent it.

**Reply by rodolfoddias** [reply]
Regarding the legacy skills, it should be easier allowing the players to mint just those skills. They could be turned into artifacts and the players could have the option to mint the legacy skills as SFT artifacts IF they want to trade them. No need to change everything to address legacy skills issue. I think it would be easier to just address it separately.
